### DISCUSSION TITLE
Number of columns does not match number of features in booster in xgboost

### DISCUSSION THREAD

**Discussion by pygabc1** [NVIDIA/NVFlare#2098] [Q&A]
Hi, I got this error message when I run federated learning xgboost, https://github.com/NVIDIA/NVFlare/tree/2.3/examples/advanced/xgboost/tree-based , with my data.    Should the number of feature of site1(client1) and the number of feature of site2 be same?  Is it caused because cross-site-validation (i.e., validating the model trained by site1 with site2 data, etc)?  If not, what caused this error?
`(2612 vs. 3734) : Number of columns does not match number of features in booster,`
where 2612 is the number of feature of site1(client1), and 3734 is the number of feature of site2.
linux OS and 3.8 python and NVFlare 2.3 are used. Thank you.

**Comment by SYangster** [comment]
This example is doing horizontal federated learning xgboost, which means that each site will have part of the whole data, while each data sample will need to contain all of the features. If you want to experiment with each of the sites containing different features, that would be supported by vertical federated xgboost (https://github.com/NVIDIA/NVFlare/tree/main/examples/advanced/vertical_xgboost).

**Reply by pygabc1** [reply]
Thank you so much. 
I run `./prepare_data.sh`  in vertical xgboost you mentioned (https://github.com/NVIDIA/NVFlare/tree/main/examples/advanced/vertical_xgboost ) to save vertical data to `/tmp/nvflare/vertical_xgb_data`.

Now when I run 
`nvflare job create -j ./jobs/vertical_xgb_psi -w psi_csv -sd ./code/psi -force`, 
I confront this issue 
` usage: nvflare [-h] [--version] {poc,preflight_check,provision,simulator,dashboard,authz_preview} ...
nvflare: error: argument sub_command: invalid choice: 'job' (choose from 'poc', 'preflight_check', 'provision', 'simulator', 'dashboard', 'authz_preview')`
linux OS and 3.8 python and NVFlare 2.3 are used.

**Reply by SYangster** [reply]
Currently you are using NVFlare 2.3, however the vertical xgboost example is a new feature on our main branch (latest branch that is not fully tested), which also includes the new job CLI commands. In order to run these new features we need to checkout the main branch and do `pip install -e .` from the root of the NVFlare repo.

**Reply by YuanTingHsieh** [reply]
@pygabc1 ^

**Reply by chesterxgchen** [reply]
The command 
```
nvflare job create -j ./jobs/vertical_xgb_psi -w psi_csv -sd ./code/psi -force
```
is only availalbe in 2.4.0 (currently in release candidate)

you can't mix 2.3.0 branch with 2.4.0 code base. 

if you want to run xgboost in 2.4.0, you can see the latest hello-world/step-by-step/higgs example series, where we have sklearn example and xgboost example notebook
